robot/client/client.go

```diff
@@ -786,6 +788,29 @@ func (rc *RobotClient) TransformPose(
 	return referenceframe.ProtobufToPoseInFrame(resp.Pose), nil
 }
 
+// TransformPointCloud will transform the pointcloud to the desired frame in the robot's frame system.
+// Do not move the robot between the generation of the initial pointcloud and the receipt
+// of the transformed pointcloud because that will make the transformations inaccurate.
+func (rc *RobotClient) TransformPointCloud(ctx context.Context, srcpc pointcloud.PointCloud, srcName, dstName string,
+) (pointcloud.PointCloud, error) {
+	// get the offset pose from a TransformPose request
+	sourceFrameZero := referenceframe.NewPoseInFrame(srcName, spatialmath.NewZeroPose())
+	transforms, err := referenceframe.LinkInFramesToTransformsProtobuf(nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := rc.client.TransformPose(ctx, &pb.TransformPoseRequest{
```

Comment:
Actually, thinking about this more, probably `GetPose` should live in the robot with the rest of the framesystem methods

Reply:
I think I will leave TransformPose here since getting the remote motion service is a little involved, but I agree about moving GetPose to the robot, potentially.